test/axes/categoryAxisTests.ts

```diff
@@ -1,180 +1,276 @@
-    axis.formatter(addPlane);
-    axis.renderTo(svg);
-    let expectedTexts = domain.map(addPlane);
-    svg.selectAll("text").each(function(d, i) {
-      let actualText = d3.select(this).text();
-      assert.strictEqual(actualText, expectedTexts[i], "formatter was applied");
+      axis.tickLabelAngle(0);
+      ticks = axis.content().selectAll("text");
+      texts = ticks[0].map((d: any) => d3.select(d).text());
+      assert.deepEqual(texts, domain, "text displayed correctly when horizontal");
+      assert.closeTo(d3.transform(axis.content().selectAll(".text-area").attr("transform")).rotate, 0,
+        window.Pixel_CloseTo_Requirement, "the ticks were rotated right");
+
+      axis.tickLabelAngle(-90);
+      ticks = axis.content().selectAll("text");
+      texts = ticks[0].map((d: any) => d3.select(d).text());
+      assert.deepEqual(texts, domain, "text displayed correctly when horizontal");
+      assert.closeTo(d3.transform(axis.content().selectAll(".text-area").attr("transform")).rotate, -90,
+        window.Pixel_CloseTo_Requirement, "the ticks were rotated left");
+
+      svg.remove();
     });
-    svg.remove();
-  });
 
-  it("width accounts for margin. innerTickLength, and padding on vertical axes", () => {
-    let svg = TestMethods.generateSVG(400, 400);
-    let xScale = new Plottable.Scales.Category().domain(["foo", "bar", "baz"]).range([400, 0]);
-    let ca = new Plottable.Axes.Category(xScale, "left");
-    ca.renderTo(svg);
+    it("re-renders with the new domain when the category scale's domain changes", () => {
+      let svg = TestMethods.generateSVG();
+      let domain = ["foo", "bar", "baz"];
+      let scale = new Plottable.Scales.Category().domain(domain);
+      let axis = new Plottable.Axes.Category(scale, "left");
+      axis.renderTo(svg);
+      let tickLabels = axis.content().selectAll(".tick-label");
+      assert.strictEqual(tickLabels.size(), domain.length, "same number of tick labels as domain entries");
+      tickLabels.each(function(d, i) {
+        let tickLabel = d3.select(this);
+        assert.strictEqual(tickLabel.text(), domain[i], "tick labels render domain");
+      });
 
-    let axisWidth = ca.width();
-    ca.tickLabelPadding(ca.tickLabelPadding() + 5);
-    assert.closeTo(ca.width(), axisWidth + 5, 2, "increasing tickLabelPadding increases width");
+      let changedDomain = ["bar", "baz", "bam"];
+      scale.domain(changedDomain);
 
-    axisWidth = ca.width();
-    ca.margin(ca.margin() + 5);
-    assert.closeTo(ca.width(), axisWidth + 5, 2, "increasing margin increases width");
+      tickLabels = axis.content().selectAll(".tick-label");
+      assert.strictEqual(tickLabels.size(), changedDomain.length, "same number of tick labels as changed domain entries");
+      tickLabels.each(function(d, i) {
+        let tickLabel = d3.select(this);
+        assert.strictEqual(tickLabel.text(), changedDomain[i], "tick labels render changed domain");
```

Comment:
I'd do a `deepEquals()` after a `map()` here, or include the index in the error message.